pswalker/iterwalk.py

```diff
@@ -98,6 +99,20 @@ def iterwalk(detectors, motors, goals, starts=None, first_steps=1,
     timeout: number, optional
         The maximum time to allow for the alignment before aborting. The
         timeout is checked after each walk step.
+
+    recovery_plan: plan, optional
+        A backup plan to run when no there is no readback on the detectors.
+        This plan should expect a complete description of the situation in the
+        form of the following keyword arguments:
+            detectors: a list of all detectors
+            motors: a list of all motors
+            goals: a list of all goals
+            detector_fields: a list of the fields that correspond to the goals
+            index: which index in these equal-length lists is active
```

Comment:
I didn't include tolerances because the goal of a recovery plan isn't to align within a tolerance, but to bring us into a state with some sort of readback on the imager. In principle though we can pass in all of the arguments and just ignore the ones we don't need for that particular recovery plan.